OSDOCS-11158: adds image mode to MicroShift install (#80599)
openshift/openshift-docs: modules/microshift-install-rhel-image-mode-csi-vg.adoc

// Module included in the following assemblies:
//
// microshift_install_bootc/microshift-install-rhel-image-mode.adoc

:_mod-docs-content-type: PROCEDURE
[id="microshift-rhel-image-mode-csi-vg_{context}"]

> = Checking the bootc image for the {microshift-short} CSI

Reviewer: ```suggestion
= Configuring the bootc image for the {microshift-short} CSI
```

> = Checking the bootc image for the {microshift-short} CSI

Reviewer: ```suggestion
= Checking the host configuration for the {microshift-short} CSI
```

If the host is already configured to have a `rhel` volume group with free space, this configuration is inherited by the container. It can be used by the {microshift-short} Logical Volume Manager (LVM) Container Storage Interface (CSI) to allocate storage.

.Prerequisites

> * A {op-system-base-full} 9.4 host with an active Red Hat subscription for building {microshift-short} bootc images and running containers.

Reviewer: ```suggestion
* A {op-system-base-full} 9.4 host for running containers.
```

* You are logged into the {op-system} 9.4 host using the user credentials that have SUDO permissions configured.

.Procedure

. Determine if the volume group exists and it has the necessary free space by running the following command:
+
[source,terminal]
----
$ sudo vgs
----
+
.Example output for existing volume group
[source,terminal]
----
  VG   #PV #LV #SN Attr   VSize   VFree
  rhel   1   1   0 wz--n- <91.02g <2.02g
----

> //Q: What does it return if there's no volume group?

Reviewer: It'll be either empty output or VFree will be 0.

> <1> If no volume group is present, the output is either empty or the `VG` value is 0.

Reviewer: ```suggestion
<1> If no free volume group is present, the output is either empty or the `VFree` value is 0.
```

> <1> If no free volume group is present, the output is either empty or the `VG` value is 0.

Reviewer: ```suggestion
<1> If no free volume group is present, the output is either empty or the `VFree` value is 0.
```

// Module included in the following assemblies:
//
// microshift_install_bootc/microshift-install-rhel-image-mode.adoc

:_mod-docs-content-type: PROCEDURE
[id="microshift-rhel-image-mode-csi-vg_{context}"]
= Checking the host configuration for the {microshift-short} CSI

> If the host is already configured to have a `rhel` volume group with free space, this configuration is inherited by the container. It can be used by the {microshift-short} Logical Volume Manager (LVM) Container Storage Interface (CSI) to allocate storage.

Reviewer: ```suggestion
If the host is already configured to have a volume group with free space, this configuration is inherited by the container. It can be used by the {microshift-short} Logical Volume Manager (LVM) Container Storage Interface (CSI) to allocate storage.
```

.Prerequisites

* A {op-system-base-full} 9.4 host with an active Red Hat subscription for building {microshift-short} bootc images and running containers.
* You are logged into the {op-system} 9.4 host using the user credentials that have `sudo` permissions.

.Procedure

. Determine if the volume group exists and it has the necessary free space by running the following command:
+
[source,terminal]
----
$ sudo vgs
----
+
.Example output for existing volume group
[source,terminal]
----
  VG   #PV #LV #SN Attr   VSize   VFree # <1>
  rhel   1   1   0 wz--n- <91.02g <2.02g
----
<1> If no free volume group is present, the output is either empty or the `VFree` value is 0.
+
[NOTE]
====

> If the host has a `rhel` volume group with the available needed storage, you can run your container now. You do not need to configure the {microshift-short} CSI further.

Reviewer: ```suggestion
If the host has a volume group with the available needed storage, you can run your container now. You do not need to configure the {microshift-short} CSI further.
```

====
+
[IMPORTANT]
====

> If you do not have a `rhel` volume group, you must set one up for the LVM CSI to allocate storage in bootc {microshift-short} containers. See the additional resources for more information.

Reviewer: ```suggestion
If you do not have a volume group, you must set one up for the LVM CSI to allocate storage in bootc {microshift-short} containers. See the additional resources for more information.
```
